File: DailyTweaks/2026-08-01-LowPowerModeHaptic/tests/test_decision.c
#include <assert.h>
#include <stdio.h>
#include "../Decision.h"

int main(void) {
    /* Initial load and disabled mode must never produce feedback. */
    assert(LPMHDecisionForChange(true, false, false, true) == LPMHHapticNone);
    assert(LPMHDecisionForChange(true, false, true, false) == LPMHHapticNone);
    assert(LPMHDecisionForChange(false, true, false, true) == LPMHHapticNone);
    assert(LPMHDecisionForChange(false, true, true, false) == LPMHHapticNone);

    /* Duplicate notifications with no state transition are rejected. */
    assert(LPMHDecisionForChange(true, true, false, false) == LPMHHapticNone);
    assert(LPMHDecisionForChange(true, true, true, true) == LPMHHapticNone);

    /* Genuine transitions select the matching haptic. */
    assert(LPMHDecisionForChange(true, true, false, true) == LPMHHapticEnabled);
    assert(LPMHDecisionForChange(true, true, true, false) == LPMHHapticDisabled);

    puts("Low Power Mode Haptic decision tests passed");
    return 0;
}

File: DailyTweaks/2026-08-03-ScreenRecordingHapticAlert/tests/test_decision.py
def should_play(enabled, initialized, previous, current):
    return enabled and initialized and previous != current

assert should_play(True, True, False, True)
assert should_play(True, True, True, False)
assert not should_play(False, True, False, True)
assert not should_play(True, False, False, True)
assert not should_play(True, True, True, True)
assert not should_play(True, True, False, False)
print('capture decision tests passed')
